Automattic/woocommerce-payments
# Update webhook processing to override existing meta data

Fixes #4771 

#### Changes proposed in this Pull Request
- Currently, we had a check for `! $order->get_meta( $key )` during webhook processing for updating metadata, which meant we only update it in case the key isn't present already for that order. This led to an issue where we might have stale data stored for order and an incoming webhook for that order wouldn't update it since the key already exists.
- This PR removes that check and thus overrides the metadata whenever there is an incoming webhook.








#### Testing instructions





* Create an order like you normally would.
* Update the metadata associated with the order, for example charge_id.
* Rerun the webhook for the `payment_intent.succeeded` event using the Stripe CLI
```
stripe events resend<EVENT_ID> --account=<ACCOUNT_ID>   
```
* Ensure the charge_id value is updated to the original value.

-------------------


**Post merge**
